plus/api/src/main/java/org/sakaiproject/plus/api/model/LineItem.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021- Charles R. Severance
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.sakaiproject.plus.api.model;
+
+import java.util.Map;
+import java.util.HashMap;
+
+import java.time.Instant;
+
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.UniqueConstraint;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.CascadeType;
+
+import org.hibernate.annotations.GenericGenerator;
+
+import org.sakaiproject.springframework.data.PersistableEntity;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "PLUS_LINEITEM",
+  indexes = { @Index(columnList = "RESOURCE_ID, CONTEXT_GUID") },
+  uniqueConstraints = { @UniqueConstraint(columnNames = { "RESOURCE_ID", "CONTEXT_GUID" }) }
+)
+
+// https://www.imsglobal.org/spec/lti-ags/v2p0#line-item-service-scope-and-allowed-http-methods
+@Getter
+@Setter
+public class LineItem extends BaseLTI implements PersistableEntity<Long> {
+
+	// This is in effect a 1-to-1 with GB_GRADABLE_OBJECT_T.ID
```

Comment:
May need to think this one through

Reply:
Yup - We should talk.  I would like to hear your ideas.